doc/classes/String.xml

```diff
@@ -181,6 +181,29 @@
 			<argument index="1" name="placeholder" type="String" default="&quot;{_}&quot;" />
 			<description>
 				Formats the string by replacing all occurrences of [code]placeholder[/code] with [code]values[/code].
+				[b]Note:[/b] There must be one and only one [code]_[/code] in the [code]placeholder[/code].
+				[b]Note:[/b] The [code]values[/code] can be a dictionary, an one-dimensional array, or a two-dimensional array. When it's a two-dimensional array, the length of the inner arrays must be [code]2[/code], or an error will be generated.
+				[b]Note:[/b] The function will iterate over the [code]values[/code]. For a dictionary, first, it will replace the [code]_[/code] with the key; Then use the result to find the first matched word; At last, use the value to replace the result (see [method replace]). For an one-dimensional array, like the dictionary, but the index number acts as the key, the element of the array acts as the value. For a two-dimensional array, the first element of the inner array acts as the key, the second one acts as the value.
+				[codeblocks]
+				[gdscript]
+				# Like this string, the placeholders may be replaced in steps.
+				var str = "The NPC, named {npc_name}, is {npc_age} years old, he comes from $1, and is a member of [organization_name]..."
+				# You may already get the npc info like this.
+				var npc_info = {"npc_name": "Tom", "npc_age": 24, "organization_id":021, "npc_equipments": {"head": 001, "left_hand": 012}}
+				# As the placeholder is the default one, so we can just format `str` like this.
+				str = str.format(npc_info) # "The NPC, named Tom, is 24 years old, he comes from $1, and is a member of [organization_name]..."
+
+				# The list of places may look like this.
+				var places = ["GodotDesert", "MountGodot", "GodotRiver", "GodotLake"]
+				# This placeholder is "$_" now.
+				str = str.format(places, "$_") # "The NPC, named Tom, is 24 years old, he comes from MountGodot, and is a member of [organization_name]..."
+
+				# May find the organization info like this.
+				var org_info = [["id", 1 ], ["organization_name", "A"], ["position", 1], ["members", []]]
+				# The placeholder now is "[_]".
+				str = str.format(org_info, "[_]") # "The NPC, named Tom, is 24 years old, he comes from MountGodot, and is a member of A..."
+				[/gdscript]
+				[/codeblocks]
```

Comment:
Since there is no C# counterpart, the GDScript-specific codeblock should be removed. Also, singular `[codeblock]` tag should be used instead:

```suggestion
				[codeblock]
				# Like this string, the placeholders may be replaced in steps.
				var str = "The NPC, named {npc_name}, is {npc_age} years old, he comes from $1, and is a member of [organization_name]..."
				# You may already get the npc info like this.
				var npc_info = {"npc_name": "Tom", "npc_age": 24, "organization_id":021, "npc_equipments": {"head": 001, "left_hand": 012}}
				# As the placeholder is the default one, so we can just format `str` like this.
				str = str.format(npc_info) # "The NPC, named Tom, is 24 years old, he comes from $1, and is a member of [organization_name]..."

				# The list of places may look like this.
				var places = ["GodotDesert", "MountGodot", "GodotRiver", "GodotLake"]
				# This placeholder is "$_" now.
				str = str.format(places, "$_") # "The NPC, named Tom, is 24 years old, he comes from MountGodot, and is a member of [organization_name]..."

				# May find the organization info like this.
				var org_info = [["id", 1 ], ["organization_name", "A"], ["position", 1], ["members", []]]
				# The placeholder now is "[_]".
				str = str.format(org_info, "[_]") # "The NPC, named Tom, is 24 years old, he comes from MountGodot, and is a member of A..."
				[/codeblock]
```

Otherwise, no code block will appear at all if the user chose to display only C# code blocks. This can be confusing.